lib/internal/Magento/Framework/Console/Cli.php

```diff
@@ -256,4 +400,125 @@ private function getCommandLoader(): Console\CommandLoader\CommandLoaderInterfac
+                if ($mode && $mode !== AppState::MODE_DEVELOPER) {
+                    return false;
+                }
+                
+                // Also check AppState as fallback
+                try {
+                    /** @var AppState $appState */
+                    $appState = $this->objectManager->get(AppState::class);
+                    return $appState->getMode() === AppState::MODE_DEVELOPER;
+                } catch (\Exception $e) {
+                    // Fallback to deployment config value
+                    return $mode === AppState::MODE_DEVELOPER;
+                }
+            }
+            
+            // Default to production (safer)
+            return false;
+        } catch (\Exception $e) {
+            // If we can't determine the mode, assume production (safer option)
+            return false;
+        }
+    }
+
+    /**
+     * Log a summary of all failed commands.
+     *
+     * @return void
+     */
+    private function logFailedCommandsSummary(): void
+    {
+        if (empty($this->failedCommands)) {
+            return;
+        }
+
+        $summary = sprintf(
+            "Failed to load %d command class(es). The CLI will continue with available commands:\n",
+            count($this->failedCommands)
+        );
+
+        foreach ($this->failedCommands as $failure) {
+            $summary .= sprintf(
+                "  - %s: %s (%s at %s)\n",
+                $failure['class'],
+                $failure['error'],
+                $failure['type'],
+                $failure['file']
+            );
+        }
+
+        $this->logger->warning($summary);
+    }
+
+    /**
+     * Log command loading error to both Magento logs and PHP error log.
+     *
+     * @param string $errorMessage
+     * @param \Exception $exception
+     * @return void
+     */
+    private function logCommandLoadingError(string $errorMessage, \Exception $exception): void
+    {
+        // Try to log to Magento's log system
+        $loggedToMagento = false;
+        if ($this->logger) {
+            try {
+                $this->logger->error($errorMessage, [
+                    'exception' => $exception,
+                    'trace' => $exception->getTraceAsString()
+                ]);
+                $loggedToMagento = true;
+                
+                // Also log a warning that CLI will continue
+                $this->logger->warning(
+                    'Some commands failed to load. The CLI will continue with available commands. ' .
+                    'Check system.log for details.'
+                );
+            } catch (\Exception $logException) {
+                // Logger failed, will show in terminal
+                $loggedToMagento = false;
+            }
+        }
+        
+        // Show actionable error in terminal (production mode)
+        $terminalMessage = "\n" . str_repeat('=', 80) . "\n";
+        $terminalMessage .= "⚠️  MAGENTO CLI ERROR (Production Mode)\n";
```

Comment:
Using emojis in error_log() messages might not be a best practice, especially if logs are parsed or consumed by external tools.